### DISCUSSION TITLE
Getting an error while configuring mastermind

### DISCUSSION THREAD

**Discussion by Nikhil-Nandagopal** [Resnovas/smartcloud#83] [Q&A] [answered]
Hi,

I've been trying to configure mastermind to add a "Actions" label to issues that are labeled as "Datasources" but I keep getting an error

This is the main.yml
https://github.com/appsmithorg/appsmith/blob/release/.github/workflows/main.yml
and the config.json
https://github.com/appsmithorg/appsmith/blob/release/.github/config.json

The error thrown by the github action
https://github.com/appsmithorg/appsmith/actions/runs/450485931

**Comment by TGTGamer** [comment]
Heya. Apologies for this issue. It looks like there's an error within my configuration for non-node JS projects. I'll open a bug now.

**Reply by Nikhil-Nandagopal** [reply]
Thanks I really love what this project does and would appreciate being able to use it for appsmith

**Reply by TGTGamer** [reply]
Bug tracking opened: https://github.com/Videndum/action-masterminds/issues/84

**Reply by TGTGamer** [reply]
Updated Release: https://github.com/Videndum/release-mastermind/releases/tag/0.0.0-alpha%2B1565
Updated Labeler: https://github.com/Videndum/label-mastermind/releases/tag/2.0.1

@Nikhil-Nandagopal

**Reply by Nikhil-Nandagopal** [reply]
@TGTGamer Thank you for the quick turn around but this still doesn't work for me
This is my latest main.yml
https://github.com/Nikhil-Nandagopal/appsmith/blob/release/.github/workflows/main.yml
and config
https://github.com/Nikhil-Nandagopal/appsmith/blob/release/.github/config.json

**Reply by TGTGamer** [reply]
> @TGTGamer Thank you for the quick turn around but this still doesn't work for me
> and config
> https://github.com/Nikhil-Nandagopal/appsmith/blob/release/.github/config.json

Please see latest specification. For easy, opened pull request on your project: https://github.com/Nikhil-Nandagopal/appsmith/pull/12

**Reply by Nikhil-Nandagopal** [reply]
@TGTGamer re-ran after merging your code and still saw the failure
https://github.com/Nikhil-Nandagopal/appsmith/actions/runs/452709308

**Reply by Nikhil-Nandagopal** [reply]
I'm using version 2.0.1

**Reply by TGTGamer** [reply]
@Nikhil-Nandagopal Once again, sorry for the issues experienced. I have found a new issue which relates to your problem. 

As I develop all the new code within [Videndum/Release Mastermind](https://github.com/videndum/release-mastermind) (./packages/release-mastermind), I use gulp to extract the changes to each of the packages. I have found an error within the gulp which caused Label Mastermind not to be updated with the changes. 

If you switch to release mastermind, you will see that it works as intended. I'm working on the fix to my gulp and will push version 2.0.2 of label mastermind this afternoon.

**Reply by Nikhil-Nandagopal** [reply]
Thanks for the quick replies @TGTGamer! I'll wait for the new package since I'm not sure how to point my action directly to release mastermind.

**Reply by TGTGamer** [reply]
You can switch between any of our sub actions and root actions by just changing which package you are using. For instance going from label mastermind would be just changing the following: 

```yml
- name: Label Mastermind
  uses: Videndum/label-mastermind@2.0.1
```

would change to
```yml
- name: Release Mastermind
  uses: Videndum/release-mastermind@0.0.0-alpha+1565
```

**Reply by TGTGamer** [reply]
@Nikhil-Nandagopal Sorry for the delay. 
New Labeler release: https://github.com/Videndum/label-mastermind/releases/tag/2.1.0

**Reply by Nikhil-Nandagopal** [reply]
@TGTGamer thank you for the quick turn around time. 2.1.0 still doesn't work for me but uses: Videndum/release-mastermind@0.0.0-alpha+1565 works fine :)

**Reply by TGTGamer** [reply]
> Thanks I really love what this project does and would appreciate being able to use it for appsmith

Heya @Nikhil-Nandagopal i've been going over some old projects, what needs changing to make this work for appsmith? Would you be able to open a new feature request to help me understand this?

**Comment by TGTGamer** [comment, marked as answer]
Fixed. Let's not talk about it. I'm feeling dumb already. 

Latest release: https://github.com/Videndum/label-mastermind/releases/tag/2.1.1

<p><em>Sourced from <a href="https://github.com/Videndum/label-mastermind/releases">Label-Mastermind releases</a>.</em></p>
<blockquote>
<h2>v2.1.1 - Fix: versioning error</h2>
<ul>
<li>I'm 100% feeling dumb. It would help if I compiled the code before releasing a new version... This version implements all previous changes from 2.0.0</li>
</ul>
<h2>v2.1.0 - Feat: Add skip label option</h2>
<ul>
<li>Fixes issue found in gulp configuration.</li>
<li>Extracts typings and index.ts</li>
<li>Adds option to skip deletion of labels</li>
<li>Adds option to disable population of context configs with shared configs</li>
</ul>
<h2>v2.0.1 - Fix: versioning</h2>
<ul>
<li>Fixes issue where if project type "other" was selected, project would fail</li>
</ul>
</blockquote>

Proof of working run: 
- Action logs: https://github.com/TGTGamer/appsmith/runs/1629590348?check_suite_focus=true#step:3:30
- Issue: https://github.com/TGTGamer/appsmith/issues/2

**Reply by Nikhil-Nandagopal** [reply]
Thanks, I can now confirm that this is working :)
